driver/src/orderbook/streamed/state.rs

```diff
@@ -79,57 +81,47 @@ impl State {
 
     /// Apply an event to the state, modifying it.
     ///
-    /// In case of error no modifications take place.
+    /// Consumes and returns back `self` because it cannot be reused in case of error.
```

Comment:
That's a cool way of enforcing you don't work with the state in case of an error.